pupil_src/shared_modules/player_methods.py

```diff
@@ -194,23 +195,124 @@ def correlate_data(data, timestamps):
+
+    def __str__(self):
+        return f"{type(self).__name__}: {super().__str__()}"
+
+
+class Pupil_Recording:
+    def __init__(self, rec_dir):
+        self._rec_dir = rec_dir
+        self._meta_info = None
+        self.load(rec_dir=rec_dir)
+
+    @property
+    def rec_dir(self):
+        return self._rec_dir
+
+    @property
+    def meta_info(self):
+        return self._meta_info
+
+    def reload(self):
+        self.load(rec_dir=self.rec_dir)
+
+    def load(self, rec_dir):
+
+        def is_video_file(file_path):
+            if not os.path.isfile(file_path):
+                return False
+            # TODO: Are there any other valid extensions?
+            valid_video_extensions = [".mp4"]
+            _, ext = os.path.splitext(file_path)
+            if ext not in valid_video_extensions:
+                return False
+            return True
+
+        if not os.path.exists(rec_dir):
+            raise InvalidRecordingException(
+                reason=f"Target at path does not exist: {rec_dir}",
+                recovery=""
+            )
+
+        if not os.path.isdir(rec_dir):
+            if is_video_file(rec_dir):
+                raise InvalidRecordingException(
+                    reason=f"The provided path is a video, not a recording directory",
+                    recovery="Please provide a recording directory"
+                )
+            else:
+                raise InvalidRecordingException(
+                    reason=f"Target at path is not a directory: {rec_dir}",
+                    recovery=""
+                )
+
+        info_path = os.path.join(rec_dir, "info.csv")
+
+        if not os.path.exists(info_path):
+            raise InvalidRecordingException(
+                reason=f"There is no info.csv in the target directory",
+                recovery=""
+            )
+
+        if not os.path.isfile(info_path):
+            raise InvalidRecordingException(
+                reason=f"Target info.csv is not a file: {info_path}",
+                recovery=""
+            )
+
+        with open(info_path, "r", encoding="utf-8") as csvfile:
+            try:
+                meta_info = csv_utils.read_key_value_file(csvfile)
+            except Exception as e:
+                raise InvalidRecordingException(
+                    reason=f"Failed reading info.csv: {e}",
+                    recovery=""
+                )
+
+        info_mandatory_keys = ["Recording Name"]
+
+        for key in info_mandatory_keys:
+            try:
+                meta_info[key]
+            except KeyError:
+                raise InvalidRecordingException(
+                    reason=f"Target info.csv does not have \"{key}\"",
+                    recovery=""
+                )
+
+        all_file_paths = glob.iglob(os.path.join(rec_dir, "*"))
+
+        # TODO: Should this validation be "are there any video files" or are there specific video files?
+        if not any(is_video_file(path) for path in all_file_paths):
```

Comment:
I think the current behavior requires at least one video but I would like to change that to not require anything but an info.csv file with start time and duration keys. See Patrick's implementation for generating wordless lookup tables.

Reply:
so what is the correct check here? only the presence of `"Start Time"` and `"Duration Time"` in `info.csv`?

I'm looking at `check_for_worldless_recording` in `update_methods.py` but there's no reference to either key. also it seems that this method is generating a `world.fake` file if a real video file is missing. `is_video_file(path)` will return `True` for any video file, including `world.fake`, so the implementation seems to still be valid.

Reply:
Let's talk this through on Thursday.